### ISSUE TITLE
`extra.vars` functionality a la `dittoSeq` 

### ISSUE THREAD

**Issue by j-andrews7** [dtm2451/dittoViz#20]
This functionality is convenient in a lot of ways. Looking at `dittoSeq`, the data collection is pretty complicated. I haven't toyed with it here, but I assume it'd just be tweaking the `.data_adjust_scatter` function to pop them on as well.

I know one can use `data.out` to get the full data, but it being embedded in the plot object was handy in its own way.

**Comment by dtm2451** [comment]
It already works to where every column is kept, and embedded:

```
example("dittoExampleData", echoe = FALSE)
scatterPlot(example_df, "gene1", "gene2", size = 2) + facet_grid(subcategory~category)
```
![image](https://github.com/dtm2451/dittoViz/assets/38870347/9535008b-5a1b-45c9-a118-2691ab519f2e)

But I've found with real data that there can be a LOT of lag from the complete lack of data trimming in scatterPlot and yPlot, so I agree that an `extra.vars` equivalent (maybe called `extra.cols`, maybe unchanged) is needed along with trimming of the unused columns.

**Comment by j-andrews7** [comment]
I may just be a dunce with ggplot, but it took me a second to figure out how the data can be accessed from the plot object itself, e.g. 

```
p <- scatterPlot(example_df, "gene1", "gene2", size = 2)
```

I tried a few approaches before finding the appropriate ones:

```
> p$data
list()
attr(,"class")
[1] "waiver"

> pg <- ggplot_build(p)
> pg$data
           x        y PANEL group shape colour size fill alpha stroke
1   2.000000 4.954196     1    -1    16  black    2   NA     1    0.5
2   2.000000 5.169925     1    -1    16  black    2   NA     1    0.5
3   2.807355 5.087463     1    -1    16  black    2   NA     1    0.5
4   3.584963 4.954196     1    -1    16  black    2   NA     1    0.5
...

> p$plot_env$Target_data
# Full df

> p$plot_env$p$layers[[1]]$data
# Full df for specific layer
```

Just commenting this here for my own sake.

**Comment by dtm2451** [comment]
mmmm, this still gives me more inspiration lol

I suppose there could be some increased consistency over whether the plot is given a default `p$data`, i.e. `ggplot(<data_given>)` versus only the geoms receiving data via equivalent of `ggplot(...) + geom_<whatever>(<data_given>)`...

The faceting in my example worked because it operates on the plot layers, but there could be some default data setup picked for all plots that is given in the top-level call, `ggplot()`, to allow easier addition of new stats or geoms!  Currently, I expect that works for some functions, but that you'd need to fish out the data from a certain layer as you do here, then give it to your `geom_<whatever>` or `stat_<whatever>` to add another geom onto a `scatterPlot()`

**Comment by j-andrews7** [comment]
Yeah, in my case, I was trying to use it for plotly stuff downstream, so when `do.hover=TRUE`, the way to access is `data <- plotly_data(p)`.
